pkg/services/ngalert/state/state.go

```diff
@@ -43,6 +43,23 @@ func (a *State) GetRuleKey() models.AlertRuleKey {
 	}
 }
 
+// ContextualState is a state bundled with contextual info from when the state originated.
+type ContextualState struct {
```

Comment:
`StateCtx`? `StateWithCtx`? `StateTransition`? Open to suggestions here.

Reply:
I like StateTransition!